### PR TITLE
Apple Pay refactor

### PR DESCRIPTION
**Pull request by mpetrenco**
### Changelog:
- Added `JPApplePayController` used to present the Apple Pay sheet and handle the delegates;
- Remade `JPApplePayService` to only make a payment transaction once a PKPayment object is obtained;
- Remade `JPApplePayWrappers` so that it actually does what it says;
- Added unit tests and fixed a bug with Apple Pay supported card networks

### CODE REVIEW

File: Source/Models/PaymentShippingMethod/JPPaymentShippingMethod.m [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//  SOFTWARE.
+
+#import "JPPaymentShippingMethod.h"
+
+@implementation JPPaymentShippingMethod
+
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                            detail:(NSString *)detail
+                             label:(nonnull NSString *)label
+                            amount:(nonnull NSDecimalNumber *)amount
```
Comment (mpetrenco) [inline Source/Models/PaymentShippingMethod/JPPaymentShippingMethod.m:32]: Should remove the `nonnull` stuff

File: JudoKit_iOSTests/Modules/PaymentMethods/Interactor/JPPaymentMethodsInteractorTest.swift [hunk 2]
```diff
@@ -56,19 +56,6 @@ class JPPaymentMethodsInteractorTest: XCTestCase {
         sut.paymentTransaction(withToken: "", andSecurityCode: nil, andCompletion: completion)
     }
     
-    /*
```
Comment (mpetrenco) [inline JudoKit_iOSTests/Modules/PaymentMethods/Interactor/JPPaymentMethodsInteractorTest.swift:59]: Should add a test for the new Apple Pay method

File: JudoKit_iOSTests/Modules/PaymentMethods/Mocks/JPPaymentMethodsInteractorMock.swift [hunk 3]
```diff
@@ -119,10 +115,9 @@ class JPPaymentMethodsInteractorMock: JPPaymentMethodsInteractor {
         return [JPPaymentMethod.card(), JPPaymentMethod.iDeal()]
     }
     
-    func startApplePay(completion: JPCompletionBlock?) {
-        let response = JPResponse()
-        completion?(response,nil)
-        startApplePay = true
+    func processApplePayment(_ payment: PKPayment,
+                             withCompletion completion: @escaping JPCompletionBlock) {
+        //TODO
```
Comment (mpetrenco) [inline JudoKit_iOSTests/Modules/PaymentMethods/Mocks/JPPaymentMethodsInteractorMock.swift:120]: Should add a mocked completion

File: JudoKit_iOSTests/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenterTest.swift [hunk 4]
```diff
@@ -174,7 +174,7 @@ class JPPaymentMethodsPresenterTest: XCTestCase {
         firstStoredCard?.isSelected = true
         JPCardStorage.sharedInstance()?.add(secondStoredCard)
         sut.handleApplePayButtonTap()
-        XCTAssertTrue(interactor.startApplePay)
+        //TODO
```
Comment (mpetrenco) [inline JudoKit_iOSTests/Modules/PaymentMethods/Presenter/JPPaymentMethodsPresenterTest.swift:177]: Should add a Payment Method presenter test

**Review by stpopa — APPROVED** [review]

**Pull request merged**